docs/dev/versions.md

```diff
@@ -0,0 +1,56 @@
+# Cirq version compatibility
+
+## Overview
+
+This document defines backwards compatibility guidelines across different versions of Cirq (either for code or data).
+
+
+## Semantic Versioning 2.0
+
+Cirq follows Semantic Versioning 2.0 ([semver](http://semver.org/)) for its public API. Each release version of Cirq has the form MAJOR.MINOR.PATCH. Changes to each number have the following meaning:
+
+
+
+*   **MAJOR**: Potentially backwards incompatible changes. Code and data that worked with a previous major release will not necessarily work with the new release. However, in most cases existing json serialized Cirq objects would be migratable to the newer release; see [Compatibility of serialized objects](#compatibility-of-serialized-objects) for details on data compatibility.
+*   **MINOR**: Backwards compatible features, speed improvements, etc. Code and data that worked with a previous minor release _and_ which depends only on the non-experimental public API will continue to work unchanged. For details on what is and is not the public API, see [What is covered](#what-is-covered).
+*   **PATCH**: Backwards compatible bug fixes.
+
+
+## What is covered
+
+Only the public APIs of Cirq are backwards compatible across minor and patch versions. The public APIs consist of
```

Comment:
since we have the "what is not covered" section, you could change the tone of this section to be more "positive". E.g. remove the leading "only"; instead of saying that code not reachable through `cirq` is not covered, say code reachable through `cirq` is covered and note that examples and dev tools aren't

Reply:
Good suggestion. Done.